c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -260,63 +291,62 @@ bool FSteam::AskChooseCategory() const
 	}
 }
 //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-bool FSteam::IsInteger(const float Float) const
+bool FSteam::IsInteger(const float& Float) const
 {
 	return (Float - floor(Float)) == 0.f;
 }
 //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-bool FSteam::IsInRange(const float Value, const int LowerBound, const int UpperBound) const
+bool FSteam::IsInRange(const float& Value, const int& LowerBound, const int& UpperBound) const
 {
 	return Value >= LowerBound && Value <= UpperBound;
 }
 //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-int FSteam::GetValidatedInput(const std::string Message, const int LowerBound, const int UpperBound) const
+void FSteam::ValidateInput(int& OutInput, const std::string& Message, const int& LowerBound, const int& UpperBound) const
 {
 	std::cout << Message;
 
-	float UserInput;
-	std::cin >> UserInput;
+	std::cin >> OutInput;
 
-	while (!std::cin.good() || !IsInteger(UserInput) || !IsInRange(UserInput, LowerBound, UpperBound))
+	while (!std::cin.good() || !IsInteger(OutInput) || !IsInRange(OutInput, LowerBound, UpperBound))
 	{
 		ResetConsoleInputScreen();
 
 		std::cout << Message;
-		std::cin >> UserInput;
+		std::cin >> OutInput;
 	}
-	return (int)UserInput;
 }
 //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-std::string FSteam::GetValidatedInput(const std::string Message) const
+void FSteam::ValidateInput(std::string& OutInput, const std::string& Message) const
 {
 	std::cout << Message;
 
-	std::string UserInput;
-	std::cin >> UserInput;
+	std::cin >> OutInput;
 
-	while (!std::cin.good() || UserInput.empty())
+	while (!std::cin.good() || OutInput.empty())
 	{
 		ResetConsoleInputScreen();
 
 		std::cout << Message;
-		std::cin >> UserInput;
+		std::cin >> OutInput;
 	}
-	return UserInput;
 }
 //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-int FSteam::GetValidDay(const int Month, const int Year) const
+const int& FSteam::GetValidDay(const int& Month, const int& Year) const
 {
+	int Day;
 	switch (Month)
```

Comment:
the empty cases can be excluded

Reply:
I don't think so, no. They are being used as shown in a lecture, that way january and march are being treated alike.